src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -1,192 +1,139 @@
-        positiionModeSparks = new FourSparkMaxWithSettings(m_frontLeft, m_rearLeft, m_frontRight, m_rearRight, frontLeftPositionSettings, rearLeftPositionSettings, frontRightPositionSettings, rearRightPositionSettings);
+        frontLeftPositionController = new SparkPositionController(frontLeftSpark, positionSettings);
+        frontRightPositionController = new SparkPositionController(frontRightSpark, positionSettings);
+        rearLeftPositionController = new SparkPositionController(rearLeftSpark, positionSettings);
+        rearRightPositionController = new SparkPositionController(rearRightSpark, positionSettings);
 
-        poseGen = new EncoderPoseGenerator(posController);
-        e_frontLeft = m_frontLeft.getEncoder();
-        e_frontRight = m_frontRight.getEncoder();
-        e_rearLeft = m_rearLeft.getEncoder();
-        e_rearRight = m_rearRight.getEncoder();
+        positionController = new SparkPositionControllerGroup(frontLeftPositionController, 
+                frontRightPositionController, rearLeftPositionController, rearRightPositionController);
 
+        reset();
     }
-
-    @Override 
-    public void periodic(){
-        logger.log("Front Left Encoder Ticks", e_frontLeft.getPosition());
-        logger.log("Front Right Encoder Ticks", e_frontRight.getPosition());
-        logger.log("Rear Left Encoder Ticks", e_rearLeft.getPosition());
-        logger.log("Rear Right Encoder Ticks", e_rearRight.getPosition());
+    public void setSpeedMode(){
+        speedSettings.configureSparkMax(frontLeftSpark);
+        speedSettings.configureSparkMax(frontRightSpark);
+        speedSettings.configureSparkMax(rearLeftSpark);
+        speedSettings.configureSparkMax(rearRightSpark);        
     }
-
-    public void reset(){
-        pc_frontLeft.resetPosition();
-        pc_frontRight.resetPosition();
-        pc_rearLeft.resetPosition();
-        pc_rearRight.resetPosition();
-        logger.log("Clicks per rotation", e_rearRight.getCountsPerRevolution());
+    public void setPositionMode(){
+        positionController.configureAll();
     }
     
+    public EncoderValues getEncoderValues(){
+       return new EncoderValues(frontLeftEncoder.getPosition(), 
+                rearLeftEncoder.getPosition(), 
+                frontRightEncoder.getPosition(), 
+                rearRightEncoder.getPosition());
+    }
+   
+    @Override 
+    public void customPeriodic(RobotPose rp, FieldPose fp) {
+        
+        logger.log("Front Left Encoder Ticks", frontLeftEncoder.getPosition());
+        logger.log("Front Right Encoder Ticks", frontRightEncoder.getPosition());
+        logger.log("Rear Left Encoder Ticks", rearLeftEncoder.getPosition());
+        logger.log("Rear Right Encoder Ticks", rearRightEncoder.getPosition());
+        
+    }
 
-    public void updatePose(PositionReader pose){
-        poseGen.updateFromOfficialPose(pose);
-        poseGen.updatePose();
+    private void localReset(){
```

Comment:
i dont understand why localrecset is called localreset. why isnt it called 'resetPosition'

Reply:
Well that is because it only resets the drive position and not navx and everything